rest/src/main/java/com/yahoo/ycsb/webservice/rest/RestClient.java

```diff
@@ -0,0 +1,370 @@
+import com.yahoo.ycsb.Status;
+import com.yahoo.ycsb.StringByteIterator;
+
+/**
+ * Class responsible for making web service requests for benchmarking purpose.
+ * Using Apache HttpClient over standard Java HTTP API as this is more flexible
+ * and provides better functionality. For example HttpClient can automatically
+ * handle redirects and proxy authentication which the standard Java API can't.
+ */
+public class RestClient extends DB {
+
+  private static final String URL_PREFIX = "url.prefix";
+  private static final String CON_TIMEOUT = "timeout.con";
+  private static final String READ_TIMEOUT = "timeout.read";
+  private static final String EXEC_TIMEOUT = "timeout.exec";
+  private static final String LOG_ENABLED = "log.enable";
+  private static final String HEADERS = "headers";
+  private static final String COMPRESSED_RESPONSE = "response.compression";
+  private boolean compressedResponse;
+  private boolean logEnabled;
+  private String urlPrefix;
+  private Properties props;
+  private String[] headers;
+  private CloseableHttpClient client;
+  private int conTimeout = 10000;
+  private int readTimeout = 10000;
+  private int execTimeout = 10000;
+  private volatile Criteria requestTimedout = new Criteria(false);
+
+  @Override
+  public void init() throws DBException {
+    props = getProperties();
+    urlPrefix = props.getProperty(URL_PREFIX, "http://127.0.0.1:8080");
+    conTimeout = Integer.valueOf(props.getProperty(CON_TIMEOUT, "10")) * 1000;
+    readTimeout = Integer.valueOf(props.getProperty(READ_TIMEOUT, "10")) * 1000;
+    execTimeout = Integer.valueOf(props.getProperty(EXEC_TIMEOUT, "10")) * 1000;
+    logEnabled = Boolean.valueOf(props.getProperty(LOG_ENABLED, "false").trim());
+    compressedResponse = Boolean.valueOf(props.getProperty(COMPRESSED_RESPONSE, "false").trim());
+    headers = props.getProperty(HEADERS, "Accept */* Content-Type application/xml user-agent Mozilla/5.0 ").trim()
+          .split(" ");
+    setupClient();
+  }
+
+  private void setupClient() {
+    RequestConfig.Builder requestBuilder = RequestConfig.custom();
+    requestBuilder = requestBuilder.setConnectTimeout(conTimeout);
+    requestBuilder = requestBuilder.setConnectionRequestTimeout(readTimeout);
+    requestBuilder = requestBuilder.setSocketTimeout(readTimeout);
+    HttpClientBuilder clientBuilder = HttpClientBuilder.create().setDefaultRequestConfig(requestBuilder.build());
+    this.client = clientBuilder.setConnectionManagerShared(true).build();
+  }
+
+  @Override
+  public Status read(String table, String endpoint, Set<String> fields, HashMap<String, ByteIterator> result) {
+    int responseCode;
+    try {
+      responseCode = httpGet(urlPrefix + endpoint, result);
+    } catch (Exception e) {
+      responseCode = handleExceptions(e, urlPrefix + endpoint, HttpMethod.GET);
+    }
+    if (logEnabled) {
+      System.out.println(new StringBuilder("GET Request: ").append(urlPrefix).append(endpoint)

```

Comment:
@risdenk these are not error messages but just information logs. That's why I didn't update them since they will carry the messages of all the HTTP codes including success like 200, 204 etc.